mailpoet/assets/js/src/email-editor/engine/hooks/use-content-validation.ts

```diff
@@ -89,14 +88,12 @@ export const useContentValidation = (): ContentValidationData => {
     rules,
   ]);
 
-  const debouncedValidateContent = useDebounce(validateContent, 500);
```

Comment:
Did you test the impact of removing the debounce? If the subscription itself is triggered reasonably perhaps we don't need the debounce, but I assume we added it because of performance reasons. We don't run the validation on every key press when writing in the editor.
There might be another workaround for the problem. Maybe we could run `select(emailEditorStore).getEditedEmailContent()` and `select(emailEditorStore).getTemplateContent()` directly in the validateContent function to get fresh data. 

> First with the invalid content and
then with the right content with the link present.
With debounce the test was not run for the content with the
link.

I would expect debounce to work the other way around. Maybe, it is related to the useCallback we use for validateContent.

Reply:
I will test the performance again. It was not an issue when I tested it. There were some more renders, but nothing too bad. The issue with the workaround you suggested is that the function would be called “before” the store has the correct data. So even with directly pulling data we may get the old values. And with debounce the function would not be called again after it is populated.

Reply:
@costasovo I tested that, and I think the performance is good enough. The `validateContent` function is not called when the notice is not displayed. If the notice is shown, it sometimes gets called too often. But I need to figure out what the alternative is. As far as I know, calling `select(emailEditorStore).getEditedEmailContent()` is impossible inside `validateContent`. For `select`, we would need to use a hook inside the function, which doesn't work, does it? We can use hooks only on the top level of components, can't we?